pkg/segment/set.go

```diff
@@ -399,12 +429,20 @@ func (s *Set) Remove(seg Iterator) GapIterator {
 		// overlap.
 		seg.SetRangeUnchecked(victim.Range())
 		seg.SetValue(victim.Value())
+		// Need to update the sibling's maxGap in this case.
+		sibling := seg.NextSegment()
+		sibling.node.updateMaxGapLeaf()
```

Comment:
It's necessary because sibling's prevgap must be affected when seg is set new value to be victim.
e.g. 
 -   1 -   3: {9 10} => 100009, maxGap: 1
 -   1: {11 12} => 100011, maxGap: 9223372036854775788
 -   2 -   0: {14 15} => 100014, maxGap: 2
 After removing seg {11 12} the maxGap of the node seg {14 15} is living in should become 4 instead of 2. But the removal of {9 10} will never update the maxGap of this node.

I update the comment to make this more clear